.github/workflows/centos.yml

```diff
@@ -0,0 +1,237 @@
+        run: |
+          source /opt/rh/devtoolset-9/enable
+          cmake3 --build build --config Release --target install -j14
+
+      - name: Zip Install Sirius
+        run: |
+          ARCHIVE_NAME="${{ steps.sirius-variables.outputs.sirius-archive-name }}"
+          zip -r $ARCHIVE_NAME "${{ steps.sirius-variables.outputs.sirius-install-dir }}"
+
+      - name: Upload sirius install
+        uses: actions/upload-artifact@v2
+        with:
+          name: centos7_sirius
+          path: ${{ steps.sirius-variables.outputs.sirius-install-path }}
+
+      - name: Publish sirius install
+        if:  ${{ github.event_name == 'release' && github.event.action == 'created' }}
+        uses: actions/upload-release-asset@v1.0.2
+        with:
+          upload_url: ${{ steps.get_release.outputs.upload_url }}
+          asset_path: ${{ steps.sirius-variables.outputs.sirius-install-path }}
+          asset_name: centos_${{ steps.sirius-variables.outputs.sirius-archive-name }}
+          asset_content_type: application/zip
+
+
+  build:
+    needs: [build_sirius]
+    name: shrd=${{ matrix.shared }} sirius=${{ matrix.sirius }} xprs=${{ matrix.xprs }} extras=${{ matrix.extras }} java=${{ matrix.java }} dotnet=${{ matrix.dotnet }} python=${{ matrix.python }}-${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    container: 'centos:centos7'
+    env:
+      XPRESS_INSTALL_DIR: xpressmp813/
+      SIRIUS_INSTALL_DIR: sirius_install/
+    strategy:
+      fail-fast: false
+      matrix:
+        xprs: [ON]
+        sirius: [ON, OFF]
+        shared: [ON, OFF]
+        extras: [OFF]
+        exclude:
+          - xprs: OFF
+            sirius: OFF
+        include:
+          - extras: OFF
+            python: OFF
+            publish-or: ON
+            java: OFF
+            dotnet: OFF
+
+    steps:
+      - name: set name variables
+        id: names
+        shell: bash
+        run: |
+          SHARED=${{ matrix.shared }}
+          [ $SHARED == "ON" ] && WITH_SHARED="_shared" || WITH_SHARED="_static"
+          XPRS=${{ matrix.xprs }}
+          [ $XPRS == "ON" ] && WITH_XPRS="_xprs" || WITH_XPRS=""
+          SIRIUS=${{ matrix.sirius }}
+          [ $SIRIUS == "ON" ] && WITH_SIRIUS="_sirius" || WITH_SIRIUS=""
+          OS="_centos7"
+          APPENDIX="${OS}${WITH_SIRIUS}${WITH_XPRS}"
+          echo "::set-output name=appendix::$APPENDIX"
+          APPENDIX_WITH_SHARED="${OS}${WITH_SHARED}${WITH_SIRIUS}${WITH_XPRS}"
+          echo "::set-output name=appendix_with_shared::$APPENDIX_WITH_SHARED"
+
+      - uses: nelonoel/branch-name@v1.0.1
+
+      - name: Get release
+        if: github.event_name == 'release' && github.event.action == 'created'
+        id: get_release
+        uses:
+          bruceadams/get-release@v1.2.3
+
+      - name: install requirements
+        run: |
+          yum install -y epel-release
+          yum install -y git redhat-lsb-core gcc gcc-c++ make wget centos-release-scl scl-utils rpm-build
```

Comment:
`gcc` and `gcc-c++` are provided by `devtoolset-9`. Therefore it is not necessary to install them here.

Reply:
ok